Products/ZenEvents/EventClassInst.py

```diff
@@ -215,7 +219,45 @@ def sendTransformException(self, eventclass, evt):
         )
         zem.sendEvent(badEvt)
 
-    def pickleFailedEvent(self, evt):     
+        transform_enabled = eventclass.transformEnabled
+        if transform_enabled:
+            disableEvt = self._updateBadTransformCounter(evt, eventclass)
+            if disableEvt:
+                disable_msg = "Transform for Event Class %s is disabled due to %d attempts to apply it: line %d, %s, " \
+                              "exception %s" % (
+                    evt.eventClass, eventclass.getProperty(MAX_TRANSFORM_FAILS), badLineNo, badLineText, exceptionText)
+                disableTransformEvt = dict(
+                    eventClass='/App/Zenoss',
+                    eventKey='transform_disabled',
+                    device=zem.host,
+                    component=evt.eventClass,
+                    summary=disable_msg,
+                    severity=4,
+                    message='Transform can be enabled again in Event Classes settings.'
+                )
+                zem.sendEvent(disableTransformEvt)
+                self.changeTransformEnabledValue(False, eventclass)
+                self.resetBadTransformCounter(evt.eventClass)
+
+    def _updateBadTransformCounter(self, evt, eventclass):
+        name = evt.eventClass
+        if not self._badtransform.get(name):
+            self._badtransform[name] = 1
+        else:
+            self._badtransform[name] += 1
+        if self._badtransform[name] >= eventclass.getProperty(MAX_TRANSFORM_FAILS):
+            return True
+        return False
+
+    def resetBadTransformCounter(self, eventclass_name):
+        self._badtransform[eventclass_name] = 0
+
+    @transact
+    def changeTransformEnabledValue(self, value, eventclass):
```

Comment:
API nitpick: `setTransformEnabled` is probably a better (not to mention, common) naming convention for setting the value of a property.